containers/unit_tests/TestWithoutInitializing.hpp

```diff
@@ -206,6 +235,42 @@ TEST(TEST_CATEGORY, resize_exec_space_dynrankview) {
   listen_tool_events(Config::DisableAll());
 }
 
+TEST(TEST_CATEGORY, realloc_exec_space_dynrankview) {
+#ifdef KOKKOS_ENABLE_CUDA
+  if (std::is_same<typename TEST_EXECSPACE::memory_space,
+                   Kokkos::CudaUVMSpace>::value)
+    GTEST_SKIP() << "skipping since CudaUVMSpace requires additional fences";
+#endif
+// FIXME_THREADS The Threads backend fences every parallel_for
+#ifdef KOKKOS_ENABLE_THREADS
+  if (std::is_same<TEST_EXECSPACE, Kokkos::Threads>::value)
+    GTEST_SKIP() << "skipping since the Threads backend isn't asynchronous";
+#endif
+
+  using namespace Kokkos::Test::Tools;
+  listen_tool_events(Config::DisableAll(), Config::EnableFences());
+  using view_type = Kokkos::DynRankView<int, TEST_EXECSPACE>;
+  view_type outer_view, outer_view2;
+
+  auto success = validate_absence(
+      [&]() {
+        view_type inner_view(Kokkos::view_alloc(TEST_EXECSPACE{}, "bla"), 8);
+        // Avoid testing the destructor
+        outer_view = inner_view;
+        Kokkos::realloc(
+            Kokkos::view_alloc(Kokkos::WithoutInitializing, TEST_EXECSPACE{}),
+            inner_view, 10);
+        outer_view2 = inner_view;
+      },
+      [&](BeginFenceEvent event) {
+        if (event.descriptor().find("HostSpace fence") != std::string::npos)
```

Comment:
Can `DynRankView` exhibit the same `"Debug Only Check for Execution Error"` that the tests for `DualView` and `ScatterView` exclude?

Reply:
The test for `View` has that exclusion, so I think we need it here, unless it's superfluous there.

Reply:
Yes, this is only necessary for View since it includes an additional case.